pkg/apiserver/rest/common.go

```diff
@@ -33,3 +39,43 @@ func sendError(ctx echo.Context, code int, message string) error {
 func sendResponse(ctx echo.Context, code int, object interface{}) error {
 	return ctx.JSON(code, object)
 }
+
+func (s *ServerImpl) GetOpenAPISpec(ctx echo.Context) error {
+	swagger, err := server.GetSwagger()
+	if err != nil {
+		return fmt.Errorf("failed to load swagger spec: %v", err)
+	}
+
+	// Use the X-Forwarded-* headers to populate the OpenAPI spec with the
+	// location where the API is being served. Using this trick the
+	// swagger-ui service dynamically loads the OpenAPI spec from the
+	// APIServer and knows where to send the TryItNow requests.
+	// Wherever the API server is accessed from through a proxy or
+	// sub-domain this will correct the servers entry to match that clients
+	// access path.
+	headers := ctx.Request().Header
+
+	log.Debugf("Got headers %#v", headers)
+
+	serverurl := &url.URL{}
+	if forwardedHost, ok := headers["X-Forwarded-Host"]; ok {
+		proto := "http"
+		if forwardedProto, ok := headers["X-Forwarded-Proto"]; ok {
+			proto = forwardedProto[0]
+		}
+		serverurl.Scheme = proto
+		serverurl.Host = forwardedHost[0]
+	}
+
+	if forwardedPrefix, ok := headers["X-Forwarded-Prefix"]; ok {
+		serverurl = serverurl.JoinPath(forwardedPrefix[0])
+	}
+
+	if *serverurl != (url.URL{}) {
```

Comment:
```suggestion
	if *serverURL != (url.URL{}) {
```

Reply:
Sam as above, for now i'm going to leave this one.